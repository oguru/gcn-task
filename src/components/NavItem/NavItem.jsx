import React from "react";
import styles from "./NavItem.module.scss";
import { Link } from "@reach/router";
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faAngleRight } from '@fortawesome/free-solid-svg-icons'

const NavItem = (props) => {
  const {handleClick, hasChildren, name, path} = props;

  if (hasChildren) {
    return (
      <li onClick={handleClick} className={styles.navItemCont}>
          <p className={styles.navItem}>{name}</p>
          <span className={styles.navArrow}>
            <FontAwesomeIcon icon={faAngleRight}/>
          </span>
        </li>
    )
  }

  return (
    <li 
      className={styles.navItemCont} 
      onClick={handleClick}
    >
      <Link
        className={styles.navItem}          
        key={name}
        // getProps={isActive}
        to={path}
      >
        {name}
      </Link>
    </li>
  )
};

export default NavItem;
