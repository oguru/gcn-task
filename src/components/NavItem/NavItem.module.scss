@import "../../variables.scss";

.navItemCont {
    border-top: solid 1px rgba(255,255,255,0.3);
    cursor: pointer;
    display: flex;
    flex-grow: 1;
    height: 30px;
    line-height: 28px;
    justify-content: space-between;
    padding: 0 $base-spacing;
    transition-duration: 200ms;
    width: 100%;

    &:hover {
        background-color: $primary-color;
    }

    .navArrow {
        color: white;
        display: flex;
        align-items: center;
        justify-content: center;
        margin-left: $base-spacing;
        height: 30px;

        svg {
           height: 20px;
           width: 20px; 
        }
    }

    .navItem {
        color: white;
        text-decoration: none;
    }
}