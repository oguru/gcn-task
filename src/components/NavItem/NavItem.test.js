import { mount } from "enzyme";
import { render, cleanup } from "@testing-library/react";
import NavItem from "./NavItem";
import React from "react";

describe("NavItem tests", () => {
  let component;
  let noChildComponent;
  const handleClick = jest.fn();
  const name = "test";
  const path = "testPath";
  
    beforeEach(() => {
      component = mount(
        <NavItem 
          handleClick={handleClick} 
          hasChildren={true} 
          name={name}
          path={path}
        />
      );
  
        noChildComponent = mount(
          <NavItem 
            handleClick={handleClick}
            hasChildren={false} 
            name={name}
            path={path}
          />
        );
    });

  afterEach(cleanup);

  it("should render", () => {
    expect(render(
      <NavItem 
        handleClick={handleClick} 
        hasChildren={true} 
        name={name}
        path={path}
      />)).toBeTruthy();

      expect(render(
        <NavItem 
          handleClick={handleClick} 
          hasChildren={false} 
          name={name}
          path={path}
        />)).toBeTruthy();
  });

  it("should match the snapshot", () => {
    expect(component).toMatchSnapshot();
    expect(noChildComponent).toMatchSnapshot();
  });

  it("should contain an li tag", () => {
    expect(component.find("li").length).toEqual(1);
    expect(noChildComponent.find("li").length).toEqual(1);
  });

  it("should render an arrow if hasChildren and none if false", () => {
    expect(component.find("FontAwesomeIcon").length).toEqual(1);
    expect(noChildComponent.find("FontAwesomeIcon").length).toEqual(0);
  });
});
