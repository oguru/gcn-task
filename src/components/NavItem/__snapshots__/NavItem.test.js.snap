// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavItem tests should match the snapshot 1`] = `ReactWrapper {}`;

exports[`NavItem tests should match the snapshot 2`] = `ReactWrapper {}`;
